src/System.Private.DataContractSerialization/src/System/Xml/XmlBaseWriter.cs

```diff
@@ -66,6 +66,14 @@ public override void Flush()
             _writer.Flush();
         }
 
+        public override Task FlushAsync()
+        {
+            if (IsClosed)
+                ThrowClosed();
```

Comment:
Basically, yes.  Our guidance has been that any usage error (meaning the developer misused the method, typically by passing incorrect arguments) can/should propagate synchronously; everything else should go out through the returned Task.